src/Symfony/Component/Messenger/Bridge/Kafka/Transport/KafkaTransportFactory.php

```diff
@@ -0,0 +1,135 @@
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Messenger\Bridge\Kafka\Transport;
+
+use function explode;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use const RD_KAFKA_RESP_ERR__ASSIGN_PARTITIONS;
+use const RD_KAFKA_RESP_ERR__REVOKE_PARTITIONS;
+use RdKafka\Conf as KafkaConf;
+use RdKafka\KafkaConsumer;
+use RdKafka\TopicPartition;
+use function sprintf;
+use function str_replace;
+use function strpos;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+use Symfony\Component\Messenger\Transport\TransportFactoryInterface;
+use Symfony\Component\Messenger\Transport\TransportInterface;
+
+/**
+ * @author Konstantin Scheumann <konstantin@konstantin.codes>
+ */
+class KafkaTransportFactory implements TransportFactoryInterface
+{
+    private const DSN_PROTOCOLS = [
+        self::DSN_PROTOCOL_KAFKA,
+        self::DSN_PROTOCOL_KAFKA_SSL,
+    ];
+    private const DSN_PROTOCOL_KAFKA = 'kafka://';
+    private const DSN_PROTOCOL_KAFKA_SSL = 'kafka+ssl://';
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(
+        ?LoggerInterface $logger
+    ) {
+        $this->logger = $logger ?? new NullLogger();
+    }
+
+    public function supports(string $dsn, array $options): bool
+    {
+        foreach (self::DSN_PROTOCOLS as $protocol) {
+            if (0 === strpos($dsn, $protocol)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public function createTransport(string $dsn, array $options, SerializerInterface $serializer): TransportInterface
+    {
+        $conf = new KafkaConf();
+
+        // Set a rebalance callback to log partition assignments (optional)
+        $conf->setRebalanceCb($this->createRebalanceCb($this->logger));
+
+        $brokers = $this->stripProtocol($dsn);
+        $conf->set('metadata.broker.list', implode(',', $brokers));
+
+        foreach (array_merge($options['topic_conf'] ?? [], $options['kafka_conf'] ?? []) as $option => $value) {
+            $conf->set($option, $value);
+        }
+
+        return new KafkaTransport(
+            $this->logger,
+            $serializer,
+            new RdKafkaFactory(),
+            new KafkaSenderProperties(
+                $conf,
+                $options['topic']['name'],
+                $options['flushTimeout'] ?? 10000,
+                $options['flushRetries'] ?? 0
+            ),
+            new KafkaReceiverProperties(
+                $conf,
+                $options['topic']['name'],
+                $options['receiveTimeout'] ?? 10000,
+                $options['commitAsync'] ?? false
+            )
+        );
+    }
+
+    private function stripProtocol(string $dsn): array
+    {
+        $brokers = [];
+        foreach (explode(',', $dsn) as $currentBroker) {
```

Comment:
sounds fragile to me. What is somebody send `kafka://host?param=option1,option2`

To provide multiple hosts I would be consistent with redis DSN in cache component